sensor/common/registry/registry_store.go

```diff
@@ -261,13 +271,15 @@ func (rs *Store) getRegistryForImageInNamespace(image *storage.ImageName, namesp
 func (rs *Store) upsertGlobalRegistry(registry string, dce config.DockerConfigEntry) error {
 	var err error
 	name := genIntegrationName(globalRegNamePrefix, "", "", registry)
-	err = rs.globalRegistries.UpdateImageIntegration(createImageIntegration(registry, dce, name))
+	_, err = rs.globalRegistries.UpdateImageIntegration(createImageIntegration(registry, dce, name))
 	if err != nil {
 		return errors.Wrapf(err, "updating registry store with registry %q", registry)
 	}
 
 	log.Debugf("Upserted global registry %q into store", registry)
 
+	metrics.SetGlobalSecretEntriesCount(rs.globalRegistries.Len())
```

Comment:
Since the global pull secret registries are stored separately wanted to have them counted separately.  With this we'll know how many 'entries' exist in the global pull secret - which may aid in troubleshooting, for example, why authentication to mirrors are not working. ([OCP docs state](https://docs.openshift.com/container-platform/4.16/post_installation_configuration/connected-to-disconnected.html#:~:text=Add%20mirror%20registry%20credential%20to%20the%20cluster%20global%20pull%2Dsecret%3A) that mirror registries should have auth added to the global pull secret)

Reply:
Ah, because this one we always have the full count avail so we 'Set' it.  The other (pull secrets) we wouldn't know the full count unless we looped through every namespace+secret+host (which would be expensive) - so instead we keep a 'running total' by increment/decrementing when each pull secret is processed.